### ISSUE TITLE
Request: add element automatically to relatedTo argument for entry/category connection

### ISSUE THREAD

**Issue by askogrand** [markhuot/craftql#74]
If you are doing a query on say what products an author was responsible for:

```graphql
query AboutPageBioEntriesQuery($section: [SectionsEnum], $slug: String) {
  entriesConnection(slug: $slug, type: Staff) {
    entries {
      __typename
      id
      slug
      title
    }
    edges {
      relatedEntries(section: $section, relatedTo: {targetElement: 72}) {
        totalCount
      }
    }
  }
}

{
  "slug": "matt-bennett",
  "section": "products"
}
```
If you add relatedTo in the relatedEntries arguments, the search will actually widen to include all entries that are relatedTo the targetElement (in this case a category). If I wanted to narrow it so that it was relatedTo the original entry and the category, Id have to manually add another element to the relatedTo array:

` relatedEntries(section: $section, relatedTo: [{targetElement: 72}, {targetElement: 160}]) {`

Where the first relation is the category, and the second is the author. Without it, the totalCount will increase, as relatedTo practically functions as orRelatedTo

**Comment by markhuot** [comment]
This is _sort of_ possible already, but it would use a slightly different syntax,

```graphql
{
  relatedEntries(section: $section, source: true) {
    totalCount
  }
}
```

This would say get me all entries related to the `edge` of the current iteration. However… the downside here is that if you try to pass additional elements to the relation it'll loose the relationship, as you're seeing. I'm working on a fix for this. I'll update you as soon as I have one.

**Comment by markhuot** [comment]
Phew, I'm struggling with the syntax here. On the one hand I like shorter, but then there's more "magic" going on and you'll have a harder time separating out `and` / `or` differences. E.g., if I blindly add `{element:{$id}}` to your `relatedTo` attribute then you have no control over whether it's the `target` or the `source` of the relationship, and that stinks…

How do you feel about these syntaxes?

First, using `-1` to signify the `edge` element? This is the most "backward friendly" of the bunch.

```graphql
relatedEntries(section: $section, relatedTo: [{targetElement: 72}, {targetElement: -1}]) {
  totalCount
}
```

Second, switching the arguments to strings and using `@` to signify the `edge`,

```graphql
relatedEntries(section: $section, relatedTo: [{targetElement: "72"}, {targetElement: "@"}]) {
  totalCount
}
```

Third, adding a new key for this sort of thing. There'd be one for `targetSelf` and one for `sourceSelf`. E.g.,

```graphql
relatedEntries(section: $section, relatedTo: [{targetElement: "72"}, {targetSelf: true}]) {
  totalCount
}
```

I'm leaning towards one of these approaches because that gives the implementer the most visibility in to what's happening. If you didn't need this level of querying you could always just use `relatedEntries` without any arguments and that would assume you mean `element:` in the `relatedTo` query.

```graphql
relatedEntries {
  totalCount
}
```

**Comment by markhuot** [comment]
FYI, `@edge` is my favorite, I think, but it's a backwards break and it I don't love the entry ids being strings…

![screen shot 2018-03-16 at 5 35 12 pm](https://user-images.githubusercontent.com/48975/37545761-80e550fa-2940-11e8-87b0-dd1212b79e1d.png)

**Comment by askogrand** [comment]
FWIW, `{targetSelf: true}` seems a nice compromise.

**Comment by markhuot** [comment]
Okay, I went with something a little more specific. This is on `dev-master` now. You can use,

```graphql
relatedEntries(relatedTo:[{targetElement:72}, {elementIsEdge:true}]){
  ...
}
```

You can also use `targetElementIsEdge:true` or `sourceElementIsEdge:true`.

Note, this is only required if you're overriding `relatedTo`. If you just want to set the `section`, for example, then you can leave this off entirely. In other words,

```graphql
relatedEntries(section:products){
  ...
}
```

is the same as,


```graphql
relatedEntries(section:products, relatedTo:{elementIsEdge:true}){
  ...
}
```

**Comment by markhuot** [comment]
This is released with 1.0.0
